.jenkins/caffe2/build.sh

```diff
@@ -119,8 +119,8 @@ if [[ $BUILD_ENVIRONMENT == *cuda* ]]; then
   export PATH="/usr/local/cuda/bin:$PATH"
 fi
 if [[ $BUILD_ENVIRONMENT == *rocm* ]]; then
-  if [[ -n "$IN_CI" ]]; then
-      # Set ROCM_ARCH to gfx900 and gfx906 for CI builds
+  if [[ -n "$IN_CI" && -z "$PYTORCH_ROCM_ARCH" ]]; then
+      # Set ROCM_ARCH to gfx900 and gfx906 for CI builds, if user doesn't override.
       echo "Limiting PYTORCH_ROCM_ARCH to gfx90[06] for CI builds"
       export PYTORCH_ROCM_ARCH="gfx900;gfx906"
```

Comment:
So I notice you use PYTORCH_ROCM_ARCH as a conditional, but then also reset it to something else. Would that ever be confusing as the user might not be expecting their input to be immediately overridden?

Reply:
IN_CI is now always 1.  It is testing for non-zero length string, so always true.  Then I'm testing for zero-length, meaning the user has not explicitly set PYTORCH_ROCM_ARCH, therefore CI job should override as it did before.

Reply:
IN_CI=1 change was made in 510334f34ba.

Reply:
What lead me to submit this PR was indeed because I had `export PYTORCH_ROCM_ARCH=gfx908` and the build.sh script was overriding what I wanted.

Reply:
Ah sorry I misread the -z